core/deployment/src/main/java/io/quarkus/deployment/pkg/PackageConfig.java

```diff
@@ -48,4 +49,17 @@
      */
     @ConfigItem(defaultValue = "-runner")
     public String runnerSuffix;
+
+    /**
+     * The output folder in which to place the output, this is resolved relative to the build
+     * systems target directory.
+     */
+    @ConfigItem
+    public Optional<String> outputDirectory;
+
+    /**
+     * The name of the final artifact
+     */
+    @ConfigItem
+    public Optional<String> outputName;
```

Comment:
I'm a bit concerned about the consequences of this on the ``Dockerfile``s we provide.

I'm more and more thinking that having some templates for these files is probably not a good idea and they shoud be generated by the build.

Do my interrogations make sense?

Reply:
I like the idea of generating them in the build, I had already been thinking about it in the context of https://github.com/quarkusio/quarkus/pull/5496

Reply:
I don't think this will actually break the dockerfile, as they seem to be based on suffix, this only changes the prefix in the same way as build.finalName in maven.